core-curriculum/knowledge/databases.json

```diff
@@ -0,0 +1,323 @@
+  "topics": [
+    {
+      "id": "database-fundamentals",
+      "name": "Database fundamentals",
+      "description": "There are many ways to store data, but being able to choose an appropriate one can be difficult.",
+      "tags": [],
+      "resources": [
+        {
+          "name": "Designing Data-Intensive Applications (Martin Kleppmann, 2017): Chapter 1 and 2",
+          "type": "book",
+          "url": "https://dataintensive.net/"
+        }
+      ],
+      "abilities": [
+        "Basic knowledge about different database types and their use cases",
+        "Know about the benefits and use cases of Relational databases",
+        "Know about the benefits and use cases of Key/Value databases",
+        "Know about the benefits and use cases of Document databases",
+        "Know about the benefits and use cases of Graph databases",
+        "Know about the benefits and use cases of Analytical databases",
+        "Considering trade-offs to when choosing between different database types"
+      ],
+      "assessment-questions": [
+        {
+          "question": "What are the most important trade-offs you need to make when choosing the right database for a project?",
+          "answer-hint": "Data model capabilities; latency vs durability; read vs write performance; CAP theorem; ACID vs BASE"
+        }
+      ]
+    },
+    {
+      "id": "rdbms-sql-basics",
+      "name": "SQL basics",
+      "description": "Relational databases and SQL",
+      "tags": [],
+      "resources": [
+        {
+          "name": "Codecademy: Learn SQL",
+          "type": "course",
+          "url": "https://www.codecademy.com/learn/learn-sql"
+        },
+        {
+          "name": "Khan Academy: Intro to SQL",
+          "type": "video",
+          "url": "https://www.khanacademy.org/computing/computer-programming/sql"
+        },
+        {
+          "name": "PostgreSQL Documentation Part I. Tutorial",
+          "type": "documentation",
+          "url": "https://www.postgresql.org/docs/10/static/tutorial.html"
+        }
+      ],
+      "abilities": [
+        "Writing basic SQL operations using INSERT, SELECT, UPDATE, DELETE",
+        "Using parameterization in queries for performance and security"
+      ],
+      "assessment-questions": [
+        {
+          "question": "What are the most commonly used SQL operations?",
+          "answer-hint": "INSERT, SELECT, UPDATE, DELETE"
+        },
+        {
+          "question": "Why is it useful to know SQL when you are using an ORM?"
+        },
+        {
+          "question": "What is an SQL injection attack and how do you prevent it?"
+        }
+      ]
+    },
+    {
+      "id": "rdbms-sql-set-operations",
+      "name": "SQL set operations",
+      "description": "Understand the concepts of set theory and how they are used in relational databases",
+      "tags": [],
+      "dependencies": [
+        "rdbms-sql-basics"
+      ],
+      "resources": [
```

Comment:
Resource contribution: http://www.sqljoins.com/sql-joins-explained-step-by-step/

Reply:
Good article, thanks! Please feel free to commit any additional resources to the branch yourself